### ISSUE TITLE
GTP-u in linux kernel

### ISSUE THREAD

**Issue by jbdamiano** [microsoft/WSL#8042]
**Is your feature request related to a problem? Please describe.**
I'm  working on GTP and I need to recompile the kernel for every release to activate the GTP in kernel
I need to do
    add the GTP-u support in kernel

diff --git a/Microsoft/config-wsl b/Microsoft/config-wsl
index ab578e1c8891..b5de4c6a7fda 100644
--- a/Microsoft/config-wsl
+++ b/Microsoft/config-wsl
@@ -1840,7 +1840,7 @@ CONFIG_IPVTAP=y
 CONFIG_VXLAN=y
 CONFIG_GENEVE=y
 # CONFIG_BAREUDP is not set
-# CONFIG_GTP is not set
+CONFIG_GTP=y
 # CONFIG_MACSEC is not set
 # CONFIG_NETCONSOLE is not set
 CONFIG_TUN=y

Is it possible to enable it by default ?
I used it since 6 months without issue.

**Comment by tyhicks** [comment]
@benhillis I'm not real familiar with this module. At first glance, some potential negatives are:

- I'm guessing that it is not something that would be widely used. It would be nice if some other users spoke up and disproved my assumption here.
- A fraction of the WSL2 users will ever use GTP-u but, with the current limitation requiring that everything be built-in to the kernel, every user would have it enabled.
  - An additional attack surface is present with no way to denylist the module
  - An additional maintenance cost when flaws/bugs are found in GTP-u

The obvious positives are that @jbdamiano wouldn't have to do this manual (re)enablement work on every WSL2 kernel release and there would be no roadblocks for future users that come along wanting to use GTP-u.

**Comment by jbdamiano** [comment]
@tyhicks The GTP-u protocol is used to encapsulate the traffic from layer 3 in Telecom data plane layer. It's needed when you are working on Telecom node.
In the kernel SCTP is activated and used also in Telecom node and I 'm guessing that it is not something that would be widely used except in Telecom business.

**Comment by jbdamiano** [comment]
With the 6.6.36.3 WSL2 from linux supports module, GTP could be activated as a module

**Comment by jbdamiano** [comment]
In addition, with this new kernel version, add the GTP in the kernel removed the module support.

**Comment by jbdamiano** [comment]
@benhillis @tyhicks I update this issue opened 2 years ago

**Comment by annis-souames** [comment]
I agree with @jbdamiano , it would be nice to have GTP kernel module as well, one use case is when experimenting with projects such as Free5GC which require GTP kernel module for certain key services to work (see [here](https://github.com/free5gc/free5gc/wiki/Installation/21bf363166eb310d05c5232345d63e0b8e753e1c)) 

But as @tyhicks already highlighted it, the use case of this module is mostly limited to telecom use cases or network engineers playing around with GTP, in most cases these would use a native Linux machine for anything serious that has to do with GTP or telco.
